File: java/listar-dominios.java
import java.io.*;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class MailgridDomainList {
    public static void main(String[] args) {
        // Definindo a URL do endpoint da API
        String url = "https://api.mailgrid.net.br/domain/list/";

        // Dados de autenticação (substitua pelo seu token real)
        String tokenAuth = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"; // Substitua pelo seu token de autenticação

        // Montando o JSON com os dados para enviar
        String jsonInputString = "{"
                + "\"token_auth\": \"" + tokenAuth + "\""
                + "}";

        try {
            // Criando um objeto URL a partir da URL do endpoint
            URL urlObj = new URL(url);
            // Abrindo uma conexão HTTP com o método POST
            HttpURLConnection connection = (HttpURLConnection) urlObj.openConnection();
            connection.setRequestMethod("POST");

            // Definindo os cabeçalhos da requisição
            connection.setRequestProperty("Content-Type", "application/json");
            connection.setDoOutput(true); // Permitindo que dados sejam enviados no corpo da requisição

            // Enviando os dados JSON para o corpo da requisição
            try (OutputStream os = connection.getOutputStream()) {
                byte[] input = jsonInputString.getBytes(StandardCharsets.UTF_8);
                os.write(input, 0, input.length); // Escreve os dados no corpo da requisição
            }

            // Obtendo o código de resposta HTTP
            int statusCode = connection.getResponseCode();

            // Lendo a resposta da API
            try (BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
                StringBuilder response = new StringBuilder();
                String line;
                while ((line = br.readLine()) != null) {
                    response.append(line);
                }

                // Verificando o código de status e exibindo a resposta
                if (statusCode >= 200 && statusCode < 300) {
                    // Se o código de status for entre 200 e 299, a requisição foi bem-sucedida
                    System.out.println("Resposta da API: ");
                    System.out.println(response.toString());
                } else {
                    // Se o código de status for maior ou igual a 400, ocorreu um erro
                    System.out.println("Erro na requisição. Código de status HTTP: " + statusCode);
                }
            }
        } catch (IOException e) {
            // Caso ocorra algum erro na requisição, exibe a mensagem de erro
            System.out.println("Erro na requisição: " + e.getMessage());
        }
    }
}

File: javascript/listar-dominios.js
// Importa o módulo https nativo do Node.js para realizar requisições HTTPS
const https = require('https');

// Definição do endpoint da API
const url = "https://api.mailgrid.net.br/domain/list/";

// Dados de autenticação (substitua pelo seu token real)
const data = JSON.stringify({
    "token_auth": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" // Token de autenticação obtido no painel do cliente
});

// Configuração da requisição HTTP
const options = {
    hostname: 'api.mailgrid.net.br', // Nome do servidor (domínio) para onde vamos enviar a requisição
    path: '/domain/list/', // Caminho (endpoint) específico da API
    method: 'POST', // Método HTTP a ser utilizado (POST)
    headers: {
        'Content-Type': 'application/json', // Tipo de conteúdo que estamos enviando, que é JSON
        'Content-Length': data.length // Comprimento do conteúdo (tamanho dos dados em bytes)
    }
};

// Realizando a requisição HTTP
const req = https.request(options, (res) => {
    // Variável para armazenar os dados da resposta
    let responseData = '';

    // A cada pedaço de dados recebido, adicionamos ao responseData
    res.on('data', (chunk) => {
        responseData += chunk;
    });

    // Quando a resposta for completamente recebida
    res.on('end', () => {
        try {
            // Tentamos converter a resposta da API para um objeto JSON
            const parsedResponse = JSON.parse(responseData);

            // Exibe a resposta formatada em JSON (bem indentado)
            console.log("Resposta da API:");
            console.log(JSON.stringify(parsedResponse, null, 2)); // JSON.stringify com indentação de 2 espaços
        } catch (error) {
            // Caso haja erro ao processar a resposta (por exemplo, se não for um JSON válido)
            console.log("Erro ao processar a resposta JSON:", error.message);
        }
    });
});

// Em caso de erro na requisição (como problemas de rede ou no servidor)
req.on('error', (error) => {
    console.error("Erro na requisição:", error.message);
});

// Envia os dados no corpo da requisição POST
req.write(data);

// Finaliza a requisição
req.end();
